pkg/webui/lib/field-description-messages.js

```diff
@@ -0,0 +1,174 @@
+  joinEuiAbsence:
+    'Contact the manufacturer or your reseller. If they can not provide a JoinEUI, and your end device is programmable, it is okay to use `0000000000000000`, but ensure that you use the same JoinEUI in your device as you enter in The Things Stack.',
+
+  appKeyDescription:
+    'An end device specific encryption key used during OTAA to derive the AppSKey (in LoRaWAN 1.1x) or both the NwkSKey and AppSKey in LoRaWAN 1.0x.',
+  appKeyLocation:
+    'It is usually pre-provisioned by the end device manufacturer, but can also be created by the user.',
+  appKeyAbsence:
+    'Contact the manufacturer or your reseller. If they can not provide an AppKey, and your end device is programmable, it is okay to generate one.',
+
+  nwkKeyDescription:
+    'A device specific encryption key used to derive the FNwkSIntKey, SNwkSIntKey, NwkSEncKey in LoRaWAN 1.1. When a LoRaWAN 1.1 capable device connects to a LoRaWAN 1.0x Network Server which does not support dual root keys (NwkKey and AppKey), the NwkKey value is used as the AppKey value.',
+  nwkKeyLocation:
+    'It is usually pre-provisioned by the end device manufacturer, but can also be created by the user.',
+  nwkKeyAbsence:
+    'Contact the manufacturer or your reseller. If they can not provide an AppKey, and your end device is programmable, it is okay to generate one.',
+
+  devIdDescription: 'A unique, human-readable identifier for your end device.',
+  devIdLocation:
+    'You make it up, so be creative. End device IDs can not be reused by multiple end devices within the same application.',
+
+  joinServerDescription:
+    "The Join Server's role is to store root keys, generate session keys, and to send those securely to the Network Server and Application Server of choice. The device contains the same root keys, which can be provisioned as part of assembly, distribution or upon installation.",
+  joinServerLocation:
+    'Contact your manufacturer or reseller to find out if your end device is pre-provisioned on an external Join Server. If not, you may use the local Join Server, or provision the end device on our Global Join Server so that you can transfer it without lock-in.',
+  joinServerAbsence:
+    'If the end device is pre-provisioned, you will need the keys from the manufacturer to activate it.',
+
+  devAddrDescription:
+    'A 32 bit non-unique identifier, assigned by the Network Server during end device activation.',
+  devAddrLocation:
+    'For OTAA, the address is created by the Network Server. If using ABP, you must create one and manually enter it in the end device and The Things Stack.',
+
+  appSKeyDescription:
+    'After activation, this encryption key is used to secure messages which carry a payload.',
+  appSKeyLocation:
+    'For OTAA, the key is generated by the Network Server. If using ABP, you must create one and manually enter it in the end device and The Things Stack.',
```

Comment:
```suggestion
    'For Activation-By-Personalization (ABP), you must generate a key and manually program it into your end device. You can use the generate button next to the input field to generate a key.',
```

This needs to be more contextual. Users seeing this tooltip will already have ABP selected. cc @benolayinka